### PR TITLE
WasabiSetupWhonix

### PR DESCRIPTION
**Pull request by davterra**
This is a guide for installing and using Wasabi in Whonix on either Qubes or VirtualBox.

I have tried to follow the "Halls Rules of Academic Writing" that nopara posted on the Slack Docs channel, in case moving towards that style becomes a "thing" here.

### CODE REVIEW

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+Install [.NET Core ${dotnetVersion} SDK](https://www.microsoft.com/net/download) for "Building Apps" in the `whonix-template`.
+
+:::tip Optional for privacy
+To disable .NET's telemetry, which sends some usage information to Microsoft:
+
+```sh
+[user@whonix-template ~]$ export DOTNET_CLI_TELEMETRY_OPTOUT=1
+```
+
```
Comment (yahiheb) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+A Wasabi-Only Whonix VM requires a very small amount of disk space.
+This allows the luxury and security option of being able to create and dedicate a Whonix Workstation VM for the exclusive purpose of safely holding the Wasabi Wallet.
+
+To avoid the risk of the wallet VM being compromised by malicious software or websites, do not install any applications in this VM other than those required by the Wasabi Wallet, or direct the Tor browser to any URL that is not necessary to install Wasabi.
+
+For an advanced layer of security, it is possible to install and run the Wasabi Wallet VM using a removable storage medium e.g., USB drive or SD card.
+This allows the wallet to be stored in a safe place away from the computer.
+
+As this is beyond the scope of this tutorial, consult the Qubes or VirtualBox user guide for details.
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
As this is beyond the scope of this tutorial, consult the Qubes, VirtualBox or Whonix user guide for details. The [Whonix documentation](https://www.whonix.org/wiki/Documentation) in particular offers multiple pages on security, hardening and privacy techniques.
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+Everything is ready to go when the small globe icons on the bottom-right side of both VMs turn green, as shown here:
+
+![](/WhonixGlobe.png)
+
+Change the Whonix user password for both VMs, as described earlier in the yellow "Caution" section.
+Then, run any needed updates for both the Gateway and Workstation VMs.
+To do so, open the terminal in both VMs by clicking on the small monitor icon on the bottom panel of each.
+
+Run the following commands in each.
+
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion

:::tip
For better security and privacy, you can [onionize](https://www.whonix.org/wiki/Onionizing_Repositories) the Debian and Whonix repositories.
:::

```
Best practice.

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+```sh
+[user@your-whonix-vm ~]$ sudo apt update
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@your-whonix-vm ~]$ sudo apt-get update
```
https://www.whonix.org/wiki/Operating_System_Software_and_Updates

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+[user@your-whonix-vm ~]$ sudo apt upgrade
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@your-whonix-vm ~]$ sudo apt-get upgrade
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+[user@your-whonix-vm ~]$ sudo apt dist-upgrade
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@your-whonix-vm ~]$ sudo apt-get dist-upgrade
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+The process for installing Wasabi from source code is the same for either Qubes or VirtualBox.
+
+First, it is necessary to install the required dependencies for compiling and running Wasabi from source.
+This will be done in the "template" Workstation VM.
+
+Start the template VM and open a terminal window.
+Then, make sure that the VM is updated.
+
+```sh
+[user@whonix-template ~]$ sudo apt update
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@whonix-template ~]$ sudo apt-get update
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+[user@whonix-template ~]$ sudo apt upgrade
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@whonix-template ~]$ sudo apt-get upgrade
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+[user@whonix-template ~]$ sudo apt dist-upgrade
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@whonix-template ~]$ sudo apt-get dist-upgrade
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+In the future, when it is necessary to update .NET Core, then do it in this VM.
+
+Now, shut down this VM.
+
+Start the Wasabi-Whonix Workstation VM
+
+Run the following commands:
+
+```sh
+[user@your-whonix-vm ~]$ sudo apt update
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@your-whonix-vm ~]$ sudo apt-get update
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+[user@your-whonix-vm ~]$ sudo apt install git
```
Comment (PulpCattel) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
[user@your-whonix-vm ~]$ sudo apt-get install git
```
https://www.whonix.org/wiki/Install_Software

**Comment by RiccardoMasutti** [comment]
README.md addition is missing.
Please merge https://github.com/davterra/WasabiDoc/pull/1

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+
+:::warning Caution
+Unlike other operating systems in which a username and password are created during the installation process, Whonix installations come with the default username: "user", and default password: "changeme".
+The default user has administrative privileges.
+
+At a minimum, the password should be changed when first starting Whonix.
+To change the password:
+
+- Open a terminal window.
+- Type: "passwd" (without the quotes) and hit 'Enter'.
```
Comment (MaxHillebrand) [inline docs/using-wasabi/WasabiSetupWhonix.md]:
```suggestion
- Type: `passwd` and hit 'Enter'.
```

File: docs/using-wasabi/WasabiSetupWhonix.md [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+```sh
+[user@source-wasabi ~]$ git remote add nopara73 https://github.com/nopara73/WalletWasabi
+[user@source-wasabi ~]$ git fetch nopara73
+[user@source-wasabi ~]$ git checkout <development branch>
+```
+
+Pull the latest commits and compile from source.
+
+```sh
+[user@source-wasabi ~]$ git pull
```
Comment (MaxHillebrand) [inline docs/using-wasabi/WasabiSetupWhonix.md]: I'm not 100% sure, but upon first build, don't you need to do `dotnet build`?

```suggestion
[user@source-wasabi ~]$ git pull
[user@source-wasabi ~]$ dotnet run
```

Comment (davterra) [reply docs/using-wasabi/WasabiSetupWhonix.md]: Oh, definitely. I must have omitted part of an attempted cut & paste from another draft.
Thanks for noticing!

**Comment by yahiheb** [comment]
I have never used `Whonix` and I don't know much about it.
@nopara73 @lontivero @molnard should this be merged

**Pull request merged**